### PR TITLE
docs: create issue template

### PR DESCRIPTION
**Pull request by abbeyperini-deque**
Create issue template for bug reports

- [ ] [view file](https://github.com/dequelabs/cauldron/blob/e22fd8866e16d570c71476fa8cbc041bb1dc73ee/.github/ISSUE_TEMPLATE/BUG.yml) to see form

Closes: #744

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/BUG.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Bug Report
+description: File a bug report
+title: "fix: "
+labels: ["bug"]
```
Comment (thuey) [inline .github/ISSUE_TEMPLATE/BUG.yml:4]:
```suggestion
labels: ["bug", "unconfirmed"]
```

File: .github/ISSUE_TEMPLATE/BUG.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      description: Also tell us, what did you expect to happen?
+      placeholder: Tell us what you see!
+      value: "A bug happened!"
+    validations:
+      required: true
+  - type: input
+    id: version
+    attributes:
+      label: Version
+      description: What version of our software are you running?
```
Comment (thuey) [inline .github/ISSUE_TEMPLATE/BUG.yml:31]:
```suggestion
      description: What version of cauldron did this happen with?
```

**Review by thuey — CHANGES_REQUESTED** [review]

**Comment by stephenmathieson** [comment]
This seems like a solution looking for a problem. Are there any examples of issues created lacking sufficient context? I do not believe this issue template will solve anything.

**Comment by abbeyperini-deque** [comment]
@stephenmathieson [@scurker suggested this](https://github.com/dequelabs/cauldron/pull/743#discussion_r950400796) in the context of Hacktoberfest/the [new homepage](https://github.com/dequelabs/cauldron/blob/docs--new-homepage/docs/Home/index.js)

**Comment by abbeyperini-deque** [comment]
This is less about solving a problem that doesn't exist and more about making a welcoming and easy to contribute to repo.

**Comment by scurker** [comment]
> This seems like a solution looking for a problem. Are there any examples of issues created lacking sufficient context? I do not believe this issue template will solve anything.

We haven't necessarily had an issue with issue creation, but marketing is wanting to potentially promote this during hacktoberfest so there will be potentially more eyes on cauldron. This is part of the over arching theme of ensuring we have clear documentation on how to contribute.

**Review by scurker — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/BUG.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      placeholder: Steps to reproduce the bug
+      value: "Steps to reproduce"
+    validations:
+      required: true
+  - type: textarea
+    id: expected-behavior
+    attributes:
+      label: Expected Behavior
+      description: What did you expect to happen?
+      placeholder: Tell us what you expected to see!
```
Comment (schne324) [inline .github/ISSUE_TEMPLATE/BUG.yml:24]:
```suggestion
      placeholder: Tell us what you expected to happen!
```

File: .github/ISSUE_TEMPLATE/BUG.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      value: "Not a bug."
+    validations:
+      required: true
+  - type: textarea
+    id: actual-behavior
+    attributes:
+      label: Actual Behavior
+      description: What actually happened?
+      placeholder: Tell us what you see!
```
Comment (schne324) [inline .github/ISSUE_TEMPLATE/BUG.yml:33]:
```suggestion
      placeholder: Tell us what you are experiencing!
```

**Comment by stephenmathieson** [comment]
> This is part of the over arching theme of ensuring we have clear documentation on how to contribute.

I don't see how adding an issue template does this, nor do the [maintainers guidelines](https://hacktoberfest.com/participation/#maintainers) from the Hacktoberfest website mention that we should add an issue template. Also, particpants don't get credit for opening issues (only PRs fixing "hacktoberfest" labelled issues).

If you all think this is worth investing the time, then go for it. I think you're wasting company time on something that will provide little/no value though.

**Comment by abbeyperini-deque** [comment]
You're not wrong, but this isn't just about Hacktoberfest. This is about making the repo marketable and friendly to contribute to in general. Hacktoberfest's suggestions are bare minimum. [Ideally, we'd also have PR templates](https://virtualcoffee.io/monthlychallenges/sept-2022), but I figured that was a no go.

**Comment by stephenmathieson** [comment]
> This is about making the repo marketable and friendly to contribute to in general.

I disagree that this is worth company time/will provide value 🤷

**Review by scurker — APPROVED** [review]

**Comment by scurker** [comment]
> I don't see how adding an issue template does this

We want people outside of deque to be able to file issues, and having a descriptive issue template if they do file issues helps reduce time if the future and sets expectations if they do so.

> If you all think this is worth investing the time, then go for it. I think you're wasting company time on something that will provide little/no value though.

IMO it's a very small time investment and is worth our time. If you have further concerns about this wasting company time, this is not the appropriate venue as this is a public repository so if there's anything further to discuss, please bring it up in either via slack or email.

**Pull request merged**